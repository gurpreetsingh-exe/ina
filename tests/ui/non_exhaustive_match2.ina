// COMPILE_FAIL

type T = T1 | T2
type V = V1(isize) | V2(bool)
type T2 = T(T, V)

fn main() {
    match T2::T(T::T1, V::V2(true)) {
        T2::T(T::T1, V::V1(i)) => {},
        T2::T(T::T2, V::V2(b)) => {},
    }
}

