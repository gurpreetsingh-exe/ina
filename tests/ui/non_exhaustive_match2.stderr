error: non-exhaustive patterns
   ╭─[tests/ui/non_exhaustive_match2.ina:8:11]
   │
 8 │     match T2::T(T::T1, V::V2(true)) {
   │           ────────────┬────────────
   │                       ╰───────────── patterns `T(T1, V2(_))` and `T(T2, V1(_))` not covered
   ∙
───╯
